### ISSUE TITLE
💅 noUndeclaredVariables fails for variables declared in typescript compiler options

### ISSUE THREAD

**Issue by Mkassabov** [biomejs/biome#1337]
### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v21.1.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUndeclaredVariables

### Playground link

https://codesandbox.io/p/sandbox/wild-hooks-4p68d9

### Expected result

Lint will not fail on `DurableObjectState` since the type is defined by `@cloudflare/workers-types` in the `tsconfig.json`'s `compilerOptions.types` property

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
We don't have the infrastructure to support this use case yet, so there isn't a quick fix **in Biome** to help you.

A workaround would be to add these types to `javascript.globals` configuration. Once we have the infrastructure, we can evaluate a better way to not trigger these false positives.
